toplevel/coqtop.ml

```diff
@@ -233,6 +233,7 @@ let init_execution opts custom_init =
   Global.set_engagement opts.config.logic.impredicative_set;
   Global.set_indices_matter opts.config.logic.indices_matter;
   Global.set_VM opts.config.enable_VM;
+  Flags.set_native_compiler (match opts.config.native_compiler with NativeOff -> false | NativeOn _ -> true);
```

Comment:
Why is there two flags?

Reply:
(I mean one flag in the environment and one as a global flag)

Reply:
They don't serve the same purpose. The global one is stronger than the environment one. The former activates all the infrastructure for native compilation, while the latter allows the kernel to use native casts in conversion. I do think that both are useful separately though.

Reply:
Just to clarify, can you give an example use case where the `env` flag is useful?

Reply:
Imagine you have a complex piece of tactic that uses `native_compute` and you want to quickly check what happens if you used vanilla kernel conversion.

Reply:
Shouldn't you simply rebind the `native_compute` tactic, then?

Reply:
I don't think you can do that due to the way Ltac works, as `native_compute` is a primitive. Also, you prove my point about the use of redefinable tactics in Ltac2.

Also also, this won't work if you want to check what happens at Qed time.

Reply:
Also, still to clarify, how is the `env` flag accessible to the user?

Reply:
Interesting: it is not! It would have sworn that there was a corresponding user-facing Goption, but my memory might be mixing it with the VM.

Reply:
So do we want to keep the flag that is essentially dead code?

Reply:
It does need to be recorded in the typing flag structure though, this allows to keep track of the option for the checker for instance, or trace actual uses of the compiler definition-wise.

Reply:
(As a matter of fact, I believe it is a feature that the kernel environment is purely functional.)

Reply:
The checker uses the flag to disable native compute (it ignores the copy stored with constants such that it's force disabled)